### PR TITLE
allow construction of arbitrary `ActorStatusKind` in tests

### PR DESCRIPTION
**Pull request by BulbaWarrior**
Not all `ActorStatusKind` variants are public, however it is sometimes desirable to construct arbitrary variants in testing environment

### CODE REVIEW

File: elfo-core/src/actor_status.rs [hunk 1]
```diff
@@ -23,13 +23,22 @@ impl ActorStatus {
...
+    #[cfg(not(feature = "test-util"))]
     const fn new(kind: ActorStatusKind) -> Self {
         Self {
             kind,
             details: None,
         }
     }
 
+    #[cfg(feature = "test-util")]
+    pub const fn new(kind: ActorStatusKind) -> Self {
```
Comment (loyd) [inline elfo-core/src/actor_status.rs:35]: What is the reason for this change? Is it to create `FAILED` and `TERMINATED` statuses in tests?

Comment (BulbaWarrior) [reply elfo-core/src/actor_status.rs:35]: yes

Comment (loyd) [reply elfo-core/src/actor_status.rs:35]: I think it's more consistent to extend the visibility of `ActorStatus::X` costs under `test-util` instead of providing another public way to create `ActorStatus.`

Comment (BulbaWarrior) [reply elfo-core/src/actor_status.rs]: fixed

File: elfo-core/src/actor_status.rs [hunk 2]
```diff
@@ -17,10 +17,17 @@ pub struct ActorStatus {
 
 impl ActorStatus {
     pub const ALARMING: ActorStatus = ActorStatus::new(ActorStatusKind::Alarming);
+
+    #[cfg(not(feature = "test-util"))]
```
Comment (BulbaWarrior) [inline elfo-core/src/actor_status.rs:21]: maybe it will look better if we add two separate impl blocks for consts under different features?
